src/amo/store.js

```diff
@@ -28,11 +29,62 @@ import redirectTo from 'core/reducers/redirectTo';
 import search from 'core/reducers/search';
 import uiState from 'core/reducers/uiState';
 import { middleware } from 'core/store';
+import type { AddonsByAuthorsState } from 'amo/reducers/addonsByAuthors';
+import type { CollectionsState } from 'amo/reducers/collections';
+import type { HomeState } from 'amo/reducers/home';
+import type { RecommendationsState } from 'amo/reducers/recommendations';
+import type { ReviewsState } from 'amo/reducers/reviews';
+import type { UserAbuseReportsState } from 'amo/reducers/userAbuseReports';
+import type { UsersState } from 'amo/reducers/users';
+import type { ViewContextState } from 'amo/reducers/viewContext';
+import type { AbuseState } from 'core/reducers/abuse';
+import type { AddonsState } from 'core/reducers/addons';
+import type { ApiState } from 'core/reducers/api';
+import type { ErrorPageState } from 'core/reducers/errorPage';
+import type { FormOverlayState } from 'core/reducers/formOverlay';
+import type { LanguageToolsState } from 'core/reducers/languageTools';
+import type { InstallationsState } from 'core/reducers/installations';
+import type { RedirectToState } from 'core/reducers/redirectTo';
+import type { SearchState } from 'core/reducers/search';
+import type { UIStateState } from 'core/reducers/uiState';
+
+export type AppState = {|
+  abuse: AbuseState,
+  addons: AddonsState,
+  addonsByAuthors: AddonsByAuthorsState,
+  api: ApiState,
+  autocomplete: Object,
+  categories: Object,
+  collections: CollectionsState,
+  errorPage: ErrorPageState,
+  errors: Object,
+  formOverlay: FormOverlayState,
+  heroBanners: Object,
+  home: HomeState,
+  infoDialog: Object,
+  installations: InstallationsState,
+  landing: Object,
+  languageTools: LanguageToolsState,
+  recommendations: RecommendationsState,
+  redirectTo: RedirectToState,
+  reviews: ReviewsState,
+  routing: Object,
+  search: SearchState,
+  uiState: UIStateState,
+  userAbuseReports: UserAbuseReportsState,
+  users: UsersState,
+  viewContext: ViewContextState,
+|};
+
+type CreateStoreParams = {|
+  history: Object,
```

Comment:
Can we do any better than `Object` for this?

Reply:
mmh this comes from a third-party lib, and we don't use flow-typed.